servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/DefaultNettyConnection.java

```diff
@@ -641,8 +642,17 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) {
                 connection.closeHandler.channelClosedOutbound(ctx);
                 connection.channelOutboundListener.channelClosed(StacklessClosedChannelException.newInstance(
                         DefaultNettyConnection.class, "userEventTriggered(ChannelOutputShutdownEvent)"));
+            } else if (evt == SslCloseCompletionEvent.SUCCESS) {
+                // Received "close_notify" alert from the peer: https://tools.ietf.org/html/rfc5246#section-7.2.1.
+                // This message notifies that the sender will not send any more messages on this connection.
+
+                // Notify close handler first to enhance error reporting and prevent LB from selecting this connection
+                connection.closeHandler.channelClosedInbound(ctx);
+                // We MUST respond with a "close_notify" alert and close down the connection immediately,
+                // discarding any pending writes.
+                connection.closeHandler.closeChannelOutbound(ctx.channel());
```

Comment:
It looks enough for me to call `closeHandler.closeChannelOutbound` here. It will generate `ChannelOutputShutdownEvent`, triggering `channelOutboundListener.channelClosed`. WDYT?

Reply:
I think `channel.close()` for h2 is fine. This will happen for the stream only, not for the parent connection. For the parent channel, I think it's netty responsibility to handle this event and fail all subsequent attempts to create new child channels.

Why should we try to flush any data if the spec says that we must discard them and close immediately? IIUC, the remote peer that generated `close_nofity` alert won't be able to ready any new data on that connection because the `SSLEngine` is already closed.

Reply:
closing the outbound makes it so that netty itself cannot followup with any fatal alerts or `close_notify` which makes debugging harder.

Reply:
`SslHandler` intercepts `close()` and emits `close_notify` alert. It was not intercepting half-closure (shoutdown output), but we handle it manually in `RequestResponseCloseHandler`.